httpexpect/datasetAPI/getDatasetDimensions_test.go

```diff
@@ -0,0 +1,211 @@
+		Database:   "datasets",
+		Collection: "dimension.options",
+		Key:        "_id",
+		Value:      "9812",
+		Update:     validAggregateDimensionsData,
+	}
+
+	docs = append(docs, datasetDoc, editionDoc, dimensionOneDoc, dimensionTwoDoc, instanceOneDoc)
+
+	d := &mongo.ManyDocs{
+		Docs: docs,
+	}
+
+	if err := mongo.TeardownMany(d); err != nil {
+		if err != mgo.ErrNotFound {
+			log.ErrorC("Was unable to run test", err, nil)
+			os.Exit(1)
+		}
+	}
+
+	if err := mongo.SetupMany(d); err != nil {
+		log.ErrorC("Was unable to run test", err, nil)
+		os.Exit(1)
+	}
+	datasetAPI := httpexpect.New(t, cfg.DatasetAPIURL)
+
+	Convey("Get a list of all dimensions of a dataset", t, func() {
+		Convey("When user is authenticated", func() {
+
+			response := datasetAPI.GET("/datasets/{id}/editions/{edition}/versions/{version}/dimensions", datasetID, edition, 1).WithHeader("internal-token", "FD0108EA-825D-411C-9B1D-41EF7727F465").
+				Expect().Status(http.StatusOK).JSON().Object()
+
+			checkDimensionsResponse(response)
+
+		})
+
+		Convey("When a user is not authenticated", func() {
+			response := datasetAPI.GET("/datasets/{id}/editions/{edition}/versions/{version}/dimensions", datasetID, edition, 1).
+				Expect().Status(http.StatusOK).JSON().Object()
+
+			response.Value("items").Array().Length().Equal(2)
+
+			checkDimensionsResponse(response)
+
+		})
+	})
+
+	if err := mongo.TeardownMany(d); err != nil {
+		if err != mgo.ErrNotFound {
+			os.Exit(1)
+		}
+	}
+}
+
+func TestGetDimensions_Failed(t *testing.T) {
+	var docs []mongo.Doc
+
+	datasetDoc := mongo.Doc{
+		Database:   "datasets",
+		Collection: "datasets",
+		Key:        "_id",
+		Value:      datasetID,
+		Update:     validPublishedDatasetData,
+	}
+
+	editionDoc := mongo.Doc{
+		Database:   "datasets",
+		Collection: "editions",
+		Key:        "_id",
+		Value:      editionID,
+		Update:     validPublishedEditionData,
+	}
+
+	docs = append(docs, datasetDoc, editionDoc)
+
+	d := &mongo.ManyDocs{
+		Docs: docs,
+	}
+
+	if err := mongo.TeardownMany(d); err != nil {
+		if err != mgo.ErrNotFound {
+			log.ErrorC("Was unable to run test", err, nil)
+			os.Exit(1)
+		}
+	}
+
+	if err := mongo.SetupMany(d); err != nil {
+		log.ErrorC("Was unable to run test", err, nil)
+		os.Exit(1)
+	}
+
+	datasetAPI := httpexpect.New(t, cfg.DatasetAPIURL)
+
+	Convey("Fail to get a list of Dimensions for a dataset", t, func() {
+		Convey("When authenticated", func() {
+			Convey("When the dataset does not exist", func() {
+				datasetAPI.GET("/datasets/{id}/editions/{edition}/versions/1/dimensions", "1234", "2018").WithHeader("internal-token", "FD0108EA-825D-411C-9B1D-41EF7727F465").
+					Expect().Status(http.StatusNotFound)
+			})
+
+			Convey("When the edition does not exist", func() {
+				datasetAPI.GET("/datasets/{id}/editions/{edition}/versions/1/dimensions", datasetID, "2018").WithHeader("internal-token", "FD0108EA-825D-411C-9B1D-41EF7727F465").
+					Expect().Status(http.StatusNotFound)
```

Comment:
Should this be a bad request? In the swagger spec it states 400 for no edition found.